tests/test_session.py

```diff
@@ -38,3 +39,7 @@ def test_default_timeout(self, mock_token):
     def test_custom_timeout(self, mock_token):
         with WorldcatSession(mock_token, timeout=1) as session:
             assert session.timeout == 1
+
+    def test_adapter(self, mock_token):
+        with WorldcatSession(mock_token) as session:
+            assert isinstance(session.adapters["https://"], HTTPAdapter)
```

Comment:
I don't think this assertion will work - if you look at `requests.Session` [here](https://github.com/psf/requests/blob/eea3bbf9ac635f465ee6c9903dc57c677952dafd/src/requests/sessions.py#L448) you will see that it has two adapters mounted by default (http & https). Your code will mount a custom adapter with retries instead of the default one (I think, haven't tested it). Above assertion will always be True.
Try something like that instead:
```python
    def test_no_retry_adapter(self, mock_token):
        with WorldcatSession(mock_token) as session:
            assert session.adapters["https://"].max_retries.total == 0

    def test_retry_adapter(self, mock_token):
        with WorldcatSession(mock_token, max_retries=3) as session:
            assert session.adapters["https://"].max_retries.total == 3
```